src/server/index.ts

```diff
@@ -101,6 +102,8 @@ initDb()
       bodyParser.urlencoded({ extended: false }),
       // application/json
       bodyParser.json(),
+      // multipart/form-data
+      fileUpload(),
```

Comment:
could you analyse the available options for Busboy here? we definitely need a maximum file size, otherwise a user can cause the server to go OOM.

Reply:
also to clarify - `bodyParser.json()` parses fields out of the `multipart/form-data` request, and `fileUpload()` parses the files?

Reply:
nope. even without `bodyParser`, `fileUpload()` will parse the fields from the `multipart/form-data` and insert them into the `req.body` object. I've verified this to be true by commenting out the `bodyParser.json()` and `bodyParser.urlencoded()` middlewares; `req.body` is still populated by the form data fields.

Reply:
I see. we should consider carefully where to place this middleware. only endpoints that ought to accept file uploads should have this middleware present i.e. the POST and PATCH methods of the API.

I was under the impression our frontend has only been sending JSON at this point - which API endpoint requires _application/x-www-form-urlencoded_?

Reply:
not too sure; after removing the urlencoded middleware, it looks like the site still works and I'm able to do basic operations like creating/editing links. will commit the removal